### PR TITLE
feat(trello): upgrade to SDK v2, add search_cards, paginate list_cards

### PR DESCRIPTION
**Pull request by ProRedCat**
Closes #337

## Summary

- **SDK v2 migration** — bumped `autohive-integrations-sdk` to `~= 2.0.0`. Every `context.fetch()` call now reads `response.data`; a shared `_unwrap_trello_response()` helper raises on non-2xx with Trello's error body so failures can't masquerade as empty success. Handlers return `ActionError(message=...)` on failure; success payloads no longer carry `result: true`/`error`. Output schemas updated accordingly.
- **New `search_cards` action** backed by `GET /1/search?modelTypes=cards`. Translates `card_name` into Trello's `name:\"...\"` operator, accepts an advanced `query`, supports `idBoards` / `idOrganizations` scoping, defaults to `is:open` (skipped when the user query already specifies `is:open|closed|archived`, including negated forms), and includes board/list on each result.
- **Paginated `list_cards`** using Trello's documented `limit` + `before`/`since` cursors (per [Atlassian's Paging guide](https://developer.atlassian.com/cloud/trello/guides/rest-api/api-introduction/#paging)). Returns a `next_before` cursor when a full page comes back so callers can walk a big board without flooding context. Default limit 50, max 1000. Compact default field set with client-side projection; `id` is always requested server-side so cursors work even when callers ask for `fields="name"`. `filter` defaults to `"open"` in code.
- **`delete_card`** now returns `{ "deleted": true, "card_id": "..." }` on success.
- **Tests** — added `trello/tests/test_trello_unit.py` (71 unit tests, mocked, CI-safe) covering: config↔handler sync, helper coercion, non-2xx → `ActionError`, every handler's success path, `list_cards` (server-side limit, `before`/`since` forwarding, `next_before` heuristics including `limit=1`, empty-page, defensive slice, `fields` without `id` still emits cursor, `fields=all` passthrough, default filter), and `search_cards` (query building, name quoting, all 4 `is:open`-injection edge cases, scoping, list/None/non-2xx body shapes, limit clamping). Expanded `test_trello_integration.py` to cover all 18 actions with live API calls (19 tests, session-scoped fixtures, self-cleaning), gated by `TRELLO_API_KEY`/`TRELLO_API_TOKEN`.

## Breaking changes (version bumped to 2.0.0)

- Handlers no longer return `{result, error}` — failures come back as `ActionError`. Output schemas were tightened accordingly.
- `list_cards` default response is now capped (`limit=50`) and uses a compact `fields` set. Callers that relied on receiving all cards in one shot must paginate via `next_before` (or pass `limit=1000`, `fields="all"`).

## Validation

```
pytest trello/tests/test_trello_unit.py
# 71 passed

pytest trello/tests/test_trello_integration.py -m integration
# 19 passed (all 18 actions covered)
```

### CODE REVIEW

**Comment by Shubhank-Jonnada** [comment]
Live integration tests and unit tests were performed — all passing (19 live, 71 unit).

File: trello/trello.py [hunk 1]
```diff
@@ -354,45 +462,180 @@ async def execute(self, inputs: Dict[str, Any], context: ExecutionContext):
...
+            # Compute next-page cursor BEFORE projection (need raw id).
+            # Trello returns cards in reverse-creation order (newest first), so
+            # the oldest card's id is the cursor to pass as `before` to fetch
+            # the next page. We only emit it when a full page came back, which
+            # is the standard "may have more" heuristic.
+            next_before = None
+            if len(cards) == limit and cards:
+                last = cards[-1]
+                if isinstance(last, dict):
+                    next_before = last.get("id")
```
[the remark this answers is not in the document]
Comment (Shubhank-Jonnada) [reply trello/trello.py:521]: this is a false positive

**Review by Shubhank-Jonnada — APPROVED** [review]

**Review by NinosMan — APPROVED** [review]
